core/dbt/exceptions.py

```diff
@@ -778,9 +778,10 @@ def package_not_found(package_name):
 
 def package_version_not_found(package_name, version_range, available_versions):
     base_msg = (
-        "Could not find a matching version for package {}\n"
+        "Could not find a matching compatible version for package {}\n"
         "  Requested range: {}\n"
-        "  Available versions: {}"
+        "  Compatible versions: {}\n"
+        "  (Not shown: versions incompatible with installed version of dbt-core)"
```

Comment:
Should we recommend using the no-version-check flag here, like we do when a dbt project fails to run due to mismatching versions?

Reply:
good idea!